reference/nonfungible-token.scilla

```diff
@@ -85,124 +99,144 @@ transition balanceOf(address: ByStr20)
   event e
 end
 
-(* @dev:    Mint new tokens. Only contractOwner can mint the token *)
-(* @param:  to      - Address of the token recipient               *)
-(* @param:  tokenId - ID of the new token minted                   *)
-(* Returns error message code_token_exist if token exists          *)
-transition transferSingle(to: ByStr20, tokenId: Uint256)
-
-  (* Sender must be the contract owner *)
-  isAuthorized = builtin eq contractOwner _sender;
-  match isAuthorized with
+(* @dev:    Mint new tokens. Only contractOwner can mint. *)
+(* @param:  to      - Address of the token recipient      *)
+(* @param:  tokenId - ID of the new token minted          *)
+(* Returns error message CodeTokenExists if token exists. *)
+(* Revert transition if invalid recipient contract.       *)
+transition mint(to: ByStr20, tokenId: Uint256)
+  (* Check if token exists *)
+  tokenExist <- exists tokenOwners[tokenId];
+  match tokenExist with
   | True =>
-    (* Check if token exists *)
-    tokenExist <- exists tokenOwnerMap[tokenId];
-    match tokenExist with
+    (* Token exists, return error code *)
+    err = CodeTokenExists;
+    MakeError err
+  | False =>
+    (* Check if sender is contractOwner *)
+    isContractOwner = builtin eq _sender contractOwner;
+    match isContractOwner with
     | True =>
-      (* Token exists, return error code *)
-      err = CodeTokenExists;
-      MakeError err
+      (* Mint new token *)
+      tokenOwners[tokenId] := to;
+      (* Add to owner's count *)
+      someCurrentCount <- ownedTokenCount[to];
+      newCount = addOwnerCount someCurrentCount;
+      ownedTokenCount[to] := newCount;
+      (* Emit success event *)
+      e = {_eventname: "MintSuccess"; by: _sender; recipient: to; token: tokenId};
+      event e;
+      (* Send dummy callback message.                                                *)
+      (* Revert transition if invalid recipient contract. Ignore if external address *)
+      msg = { _tag : "transferCallBack"; _recipient : to; _amount : Uint128 0 };
```

Comment:
Both are `mint` and `transfer` currently doing a "transfer" to, I think in the case of handling such cases in the subsequent contract, there is no different I guess? It will be easier if I just create 1 `approvedSpender` transition in the subsequent contract to handle both scenarios (minting and transferring).